src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -563,6 +562,25 @@ private Symbol ResolveMultipleSymbolsInAttributeTypeLookup(ArrayBuilder<Symbol>
             return null;
         }
 
+        private static bool NameAndArityMatchRecursively(Symbol x, Symbol y)
+        {
+            if (isRoot(x))
+            {
+                return isRoot(y);
+            }
+            if (isRoot(y))
+            {
+                return false;
+            }
+            if (!NameAndArityMatchRecursively(x.ContainingSymbol, y.ContainingSymbol))
```

Comment:
nit: consider swapping the name comparison with the recursive call, so that we fail faster when there are differences.